hugolib/page.go

```diff
@@ -2007,24 +2007,16 @@ func sectionsFromDir(dirname string) []string {
 	return strings.Split(dirname, helpers.FilePathSeparator)
 }
 
-const (
-	regularPageFileNameDoesNotStartWith = "_index"
-
-	// There can be "my_regular_index_page.md but not /_index_file.md
-	regularPageFileNameDoesNotContain = helpers.FilePathSeparator + regularPageFileNameDoesNotStartWith
-)
-
-func kindFromFilename(filename string) string {
-	if !strings.HasPrefix(filename, regularPageFileNameDoesNotStartWith) && !strings.Contains(filename, regularPageFileNameDoesNotContain) {
-		return KindPage
-	}
-
-	if strings.HasPrefix(filename, "_index") {
+func kindFromFileInfo(fi *fileInfo) string {
+	if fi.TranslationBaseName() == "_index" {
+		if len(fi.Dir()) > 0 {
```

Comment:
This patch cleans it up nicely, but to check for non-empty strings I think `if fi.Dir() != ""` is better. Looking at your code, while correct, makes me start to think about "what is the importance of the length being greater than something?" `if len(fi.Dir()) != 0` is my second choice, but I prefer the first one.

Reply:
I agree! Coming right up!